src/app/components/core/configuration-manager/configuration-manager.component.ts

```diff
@@ -40,92 +38,57 @@ export class ConfigurationManagerComponent implements OnInit {
-                displayName: element.displayName,
-                description: element.description
-              });
-            } else {
-              this.rootCategories.push({
-                key: element.key,
-                description: element.description
-              });
-            }
-            this.rootCategories = this.rootCategories.filter(el => el.key.toUpperCase() !== 'SOUTH')
-              .filter(el => el.key.toUpperCase() !== 'NORTH')
-              .filter(el => el.key.toUpperCase() !== 'NOTIFICATIONS');
-          });
-          if (onLoadingPage === true) {
-            this.getChildren(this.selectedRootCategory);
-          }
-        },
-        error => {
-          if (error.status === 0) {
-            console.log('service down ', error);
-          } else {
-            this.alertService.error(error.statusText);
-          }
-        });
-  }
-
-  getSelectedCategoryConfig(rootCategory: any) {
-    this.selectedRootCategory = this.hasProperty(rootCategory, 'displayName') === true ?
-      rootCategory.displayName : rootCategory.key;
-    this.getChildren(rootCategory.key);
-  }
-
-  public getChildren(categoryName) {
+  public getTreeStructure() {
     /** request started */
     this.ngProgress.start();
     this.tree.treeModel.nodes = [];
 
-    this.nodes = [];
-    this.configService.getChildren(categoryName).
+    let _nodes = [];
+    this.configService.getRootCategories().
       subscribe(
         (data) => {
           /** request completed */
           this.ngProgress.done();
-          const rootCategories = this.rootCategories.filter(el => el.key === categoryName);
 
           // Check if there is any category
-          if (rootCategories.length > 0 && data['categories'].length === 0) {
+          if (data['categories'].length === 0) {
             this.isChild = false;
-            this.getCategory(rootCategories[0].key, rootCategories[0].description);
             this.categoryData = [];
             return;
           }
           this.isChild = true;
           data['categories'].forEach(element => {
-            if (element.hasOwnProperty('displayName')) {
-              this.nodes.push({
-                id: element.key,
-                name: element.displayName,
-                description: element.description,
-                hasChildren: true, children: []
-              });
-            } else {
-              this.nodes.push({
-                id: element.key,
-                name: element.description,
-                description: element.description,
-                hasChildren: true, children: []
-              });
+            _nodes.push({
+              id: element.key,
+              name: (element.hasOwnProperty('displayName')) ? element.displayName : element.description,
+              description: element.description,
+              hasChildren: (element.children.length) > 0 ? true : false , children: this.addNameKey(element.children)
```

Comment:
move children key value to next line after

`hasChildren: element.children.length > 0,`